src/auth/auth-api-request.ts

```diff
@@ -786,21 +859,58 @@ export class FirebaseAuthRequestHandler {
       request.localId = request.uid;
       delete request.uid;
     }
-    return this.invokeRequestHandler(FIREBASE_AUTH_SIGN_UP_NEW_USER, request)
+    return this.invokeRequestHandler(this.authUrlBuilder, FIREBASE_AUTH_SIGN_UP_NEW_USER, request)
       .then((response: any) => {
         // Return the user id.
         return response.localId as string;
       });
   }
 
+  /**
+   * Generates the out of band email action link for the email specified using the action code settings provided.
+   * Returns a promise that resolves with the generated link.
+   *
+   * @param {string} requestType The request type. This could be either used for password reset,
+   *     email verification, email link sign-in.
+   * @param {string} email The email of the user the link is being sent to.
+   * @param {ActionCodeSettings=} actionCodeSettings The optional action code setings which defines whether
+   *     the link is to be handled by a mobile app and the additional state information to be passed in the
+   *     deep link, etc.
+   * @return {Promise<string>} A promise that resolves with the email action link.
+   */
+  public getEmailActionLink(
+      requestType: string, email: string,
+      actionCodeSettings?: ActionCodeSettings): Promise<string> {
+    let request = {requestType, email, returnOobLink: true};
+    // ActionCodeSettings required for email link sign-in to determine the url where the sign-in will
```

Comment:
This will check it. There is a unit test to confirm this case.
The condition: `if (typeof actionCodeSettings !== 'undefined' || requestType === 'EMAIL_SIGNIN')`
will trigger for email link sign in and `new ActionCodeSettingsBuilder(actionCodeSettings)` will throw if no actioncodesettings is provided.